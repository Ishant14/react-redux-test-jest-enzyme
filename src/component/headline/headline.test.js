import React from 'react';
import Headline from './index';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';


describe("Header Component ", () => {
    test("it should render properly ", () => {
        const component = renderer.create(<Headline header="Post" desc="test"/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test("it should not render with no props", () => {
        const component = shallow(<Headline />);
        console.log(component.debug());
        const h1 = component.find("h1");
        
        expect(h1.length).toBe(0);
    })

    test("it should not render with no props", () => {
        const component = shallow(<Headline header="Post" desc="test"/>);
        const h1 = component.find("h1");
        expect(h1.length).toBe(1);
    })
   

})