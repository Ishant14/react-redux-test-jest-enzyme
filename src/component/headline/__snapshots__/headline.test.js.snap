// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component  it should render properly  1`] = `
<div>
  <h1>
    Post
  </h1>
  <p>
    test
  </p>
</div>
`;
